sc-memory/tests/sc-memory/common/sc-elements.cpp

```diff
@@ -48,3 +48,47 @@ TEST_F(ScMemoryTest, elements)
   EXPECT_FALSE(ctx.IsElement(node));
   EXPECT_FALSE(ctx.IsElement(edge));
 }
+
+TEST_F(ScMemoryTest, CreateDeleteCountEdges)
+{
+  ScMemoryContext ctx(sc_access_lvl_make_min, "CreateDeleteCountEdges");
+
+  ScAddr const node = ctx.CreateNode(ScType::Const);
+  EXPECT_TRUE(node.IsValid());
+
+  ScAddr const link = ctx.CreateLink();
+  EXPECT_TRUE(link.IsValid());
+
+  EXPECT_EQ(ctx.GetElementOutputArcsCount(node), 0u);
+  EXPECT_EQ(ctx.GetElementInputArcsCount(node), 0u);
+  EXPECT_EQ(ctx.GetElementOutputArcsCount(link), 0u);
+  EXPECT_EQ(ctx.GetElementInputArcsCount(link), 0u);
+
+  ScAddr edge = ctx.CreateEdge(ScType::EdgeAccessConstPosPerm, node, link);
+  EXPECT_TRUE(edge.IsValid());
+  EXPECT_EQ(ctx.GetElementOutputArcsCount(node), 1u);
+  EXPECT_EQ(ctx.GetElementInputArcsCount(node), 0u);
+  EXPECT_EQ(ctx.GetElementOutputArcsCount(link), 0u);
+  EXPECT_EQ(ctx.GetElementInputArcsCount(link), 1u);
+
+  edge = ctx.CreateEdge(ScType::EdgeAccessConstPosPerm, node, link);
+  EXPECT_TRUE(edge.IsValid());
+  EXPECT_EQ(ctx.GetElementOutputArcsCount(node), 2u);
+  EXPECT_EQ(ctx.GetElementInputArcsCount(node), 0u);
+  EXPECT_EQ(ctx.GetElementOutputArcsCount(link), 0u);
+  EXPECT_EQ(ctx.GetElementInputArcsCount(link), 2u);
+
+  EXPECT_TRUE(ctx.EraseElement(edge));
+  EXPECT_TRUE(edge.IsValid());
+  EXPECT_EQ(ctx.GetElementOutputArcsCount(node), 1u);
+  EXPECT_EQ(ctx.GetElementInputArcsCount(node), 0u);
+  EXPECT_EQ(ctx.GetElementOutputArcsCount(link), 0u);
+  EXPECT_EQ(ctx.GetElementInputArcsCount(link), 1u);
+
+  edge = ctx.CreateEdge(ScType::EdgeAccessConstPosPerm, node, link);
+  EXPECT_TRUE(edge.IsValid());
+  EXPECT_EQ(ctx.GetElementOutputArcsCount(node), 2u);
+  EXPECT_EQ(ctx.GetElementInputArcsCount(node), 0u);
+  EXPECT_EQ(ctx.GetElementOutputArcsCount(link), 0u);
+  EXPECT_EQ(ctx.GetElementInputArcsCount(link), 2u);
```

Comment:
```suggestion
  EXPECT_EQ(ctx.GetElementInputArcsCount(link), 2u);
  
    ScAddr const relation = ctx.CreateNode(ScType::NodeConstRole);
  EXPECT_TRUE(relation.IsValid());
  ScAddr edge2 = ctx.CreateEdge(ScType::EdgeAccessConstPosPerm, relation, edge);
  EXPECT_TRUE(edge2.IsValid());
  EXPECT_EQ(ctx.GetElementOutputArcsCount(relation), 1u);
  EXPECT_EQ(ctx.GetElementInputArcsCount(relation), 0u);
  EXPECT_EQ(ctx.GetElementOutputArcsCount(edge), 0u);
  EXPECT_EQ(ctx.GetElementInputArcsCount(edge), 1u);

  EXPECT_TRUE(ctx.EraseElement(edge));
  EXPECT_EQ(ctx.GetElementInputArcsCount(edge), 0u);
  EXPECT_EQ(ctx.GetElementOutputArcsCount(edge), 0u);
  EXPECT_EQ(ctx.GetElementOutputArcsCount(node), 1u);
  EXPECT_EQ(ctx.GetElementInputArcsCount(node), 0u);
  EXPECT_EQ(ctx.GetElementOutputArcsCount(link), 0u);
  EXPECT_EQ(ctx.GetElementInputArcsCount(link), 1u);
  EXPECT_EQ(ctx.GetElementOutputArcsCount(relation), 0u);
  EXPECT_EQ(ctx.GetElementInputArcsCount(relation), 0u);
```
I tested with by this, it's ok. But I am not sure about response of GetElementOutputArcsCount for erased elemets

Reply:
You can't get output/input arcs count for deleted edged

Reply:
So, this test was passed during my review